From d2fa1fc74f4434c193e6e79ae64f9c8b581b6ad0 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <adrian.alonso@nxp.com>
Date: Thu, 28 Apr 2016 14:02:03 -0500
Subject: [PATCH 1031/5242] MLK-12733: ARM: imx: imx6q: lppdr2 fix audio freq
 mode

commit  340db794d115c12950618c5858d336ae0b0977f1 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix Audio frequency operation mode (100Mhz), wrong branch
condition was causing to switch to high speed timings settings
if (freq mode <= 100Mhz) use low speed timings settings

On Audio freq mode request
1. Set timings to operate on low speed
2. Switch mmdc clock root to 100Mhz

Signed-off-by: Adrian Alonso <adrian.alonso@nxp.com>
Signed-off-by: Ranjani Vaidyanathan <ranjani.vaidyanathan@nxp.com>
(cherry-picked from commit 79756d9defae46b67cd37807b859b52e63b18496)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/lpddr2_freq_imx6q.S |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/lpddr2_freq_imx6q.S b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
index 3eede66..2e3b2f3 100644
--- a/arch/arm/mach-imx/lpddr2_freq_imx6q.S
+++ b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
@@ -626,7 +626,7 @@ done:
 
 	ldr	r10,=100000000
 	cmp	r0, r10
-	blt	skip_mmdc_clk_check
+	ble	skip_mmdc_clk_check
 	mmdc_clk_above_100MHz
 
 skip_mmdc_clk_check:
-- 
1.7.9.5

